wallets/client/components/passphrase.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react'
+import { CopyButton } from '@/components/form'
+import { QRCodeSVG } from 'qrcode.react'
+import styles from '@/styles/wallet.module.css'
+import { useWallets } from '@/wallets/client/context'
+import { useGenerateRandomKey, useWalletEncryptionUpdate } from '@/wallets/client/hooks'
+import useEffectOnce from '@/components/use-effect-once'
+
+export function Passphrase () {
+  const [passphrase, setPassphrase] = useState(null)
+  const wallets = useWallets()
+  const generateRandomKey = useGenerateRandomKey()
+
+  const updateWalletEncryption = useWalletEncryptionUpdate()
+
+  useEffectOnce(() => {
+    async function updateKey () {
+      const { passphrase, key, hash } = await generateRandomKey()
+      await updateWalletEncryption({ key, hash, wallets })
+      setPassphrase(passphrase)
+    }
+    updateKey()
+  }, [])
```

Comment:
`useEffect` in strict mode (which is only on in dev) runs twice, so you can detect when you're doing things you should not be doing in `useEffect`.

Reply:
Mhh yes, but I think strict mode was messing with development too much here.

But I don't remember if it broke something or it was just an unnecessary API call.
Should I not generate the new passphrase and run the migration in an `useEffect`?

> like doing something that isn't idempotent.

I guess I could generate the passphrase and run the wallet encryption in the `onShow` callback of `useShowPassphrase` and then just pass `passphrase` to the `Passphrase` component.

Reply:
Anything that doesn't require this hack, and isn't a hack itself, is probably better. I think that's why strict mode exists - prevent devs from doing stuff based on the component lifecycle that they shouldn't be doing.